### DISCUSSION TITLE
Help find style settings

### DISCUSSION THREAD

**Discussion by ghost** [koreader/koreader#11861] [Q&A] [answered]
Please somebody, tell me how to disable the built-in publisher's indents at the beginning of each paragraph? Otherwise they are too big, the indentation is too large. I looked through all the style settings and couldn't find it...
The photo is attached to better understanding.

![IMG_20240520_195045.jpg](https://github.com/koreader/koreader/assets/168462969/1b42b9aa-73aa-4a4b-961a-17a78d74228d)

**Comment by mergen3107** [comment]
Book - second tab - Style Tweaks - Book-specific tweaks.
Type in the following code:
```
p {
    text-indent: 0 !important;
}
```

**Reply by ghost** [reply]
> Book - second tab - Style Tweaks - Book-specific tweaks.
> Type in the following code:
> ```
> p {
>     text-indent: 0 !important;
> }
> ```

Thanks, it worked. But...
The changes are small. The indentation has decreased, but only by a centimeter, and can't it be reduced any more? It seems that other books have even less indentation.

**Reply by mergen3107** [reply]
That's weird :D 
As a nuke option, try this for now:
Bottom menu - last tab - uncheck "Embedded style"
If this works, then highlight a few words in a any first line and tap "View HTML". Take a screenshot or photo and share here

**Reply by ghost** [reply]
> That's weird :D 
> As a nuke option, try this for now:
> Bottom menu - last tab - uncheck "Embedded style"
> If this works, then highlight a few words in a any first line and tap "View HTML". Take a screenshot or photo and share here 

![IMG_20240520_201847.jpg](https://github.com/koreader/koreader/assets/168462969/0eecf590-14d6-4ca1-8754-8bc5552c5276)

I disabled the embedded style, but it didn't work anyway. But I can show you the html of this indentation, when I turned off the embedded style: (higher)

**Reply by mergen3107** [reply]
Ok, so it is not just p then. Try adding calibre4, calibre 2, calibre, one by one. 

it will look like (for example):

p, calibre4 {

But you said it didn't go away after you disabled them all... 
so maybe it is in top menu - second tab - style tweaks? Have a look at all the options, one of them should be responsible for text indentation

**Reply by ghost** [reply]
> Ok, so it is not just p then. Try adding calibre4, calibre 2, calibre, one by one. 
> 
> it will look like (for example):
> 
> p, calibre4 {
> 
> But you said it didn't go away after you disabled them all... 
> so maybe it is in top menu - second tab - style tweaks? Have a look at all the options, one of them should be responsible for text indentation

Please look down at other answer from poire-z, he solved my problem, but I want it in file format 🥺 like userpatch or user style tweak, to put in the folder and activate it by tap in another similar book 🙏

**Comment by poire-z** [comment, marked as answer]
You'll find some tweaks (like the one suggested, without the need to type it) under
Book - second tab - `Style tweaks > Pagagraphs > Paragraph first-line indentation>`
This targets normal paragraph, but it's possible your publisher/book use other tricks.

Seeing your screenshot, it seems the book uses:
` <span class="calibre4">    (lots of hardcoded space/no-break-spaces...)   </span>`
to uglily do that indentation...
It's really your publisher/book explicitely doing that, using spaces as part of the text: it's not something we can fix with styles. It's a crappy book, that KOReader can't solve: it is rendering the content that is given.

But there may be a trick, if really ALL the `<p class="Style">` start with such a  `<span class="calibre4">    (lots of hardcoded space/no-break-spaces...)   </span>`: you could hide them, with a book style tweak like:

`p.Style > span.calibre4:first-child { display: none !important; }` (not tested, I may have screwed it up)

But this will also hide any first <span style="calibre4"> that may have real content...

**Reply by ghost** [reply]
> You'll find some tweaks (like the one suggested, without the need to type it) under
> Book - second tab - `Style tweaks > Pagagraphs > Paragraph first-line indentation>`
> This targets normal paragraph, but it's possible your publisher/book use other tricks.
> 
> Seeing your screenshot, it seems the book uses:
> ` <span class="calibre4">    (lots of hardcoded space/no-break-spaces...)   </span>`
> to uglily do that indentation...
> It's really your publisher/book explicitely doing that, using spaces as part of the text: it's not something we can fix with styles. It's a crappy book, that KOReader can't solve: it is rendering the content that is given.
> 
> But there may be a trick, if really ALL the `<p class="Style">` start with such a  `<span class="calibre4">    (lots of hardcoded space/no-break-spaces...)   </span>`: you could hide them, with a book style tweak like:
> 
> `p.Style > span.calibre4:first-child { display: none !important; }` (not tested, I may have screwed it up)
> 
> But this will also hide any first <span style="calibre4"> that may have real content...
> 

THANK YOU VERY MUCH. IT WORKED.

But I wanted to ask, can u maybe write this to a patch or how u call this... Style tweak? So I can put it in some folder and activate it in another similar book just by tap. Please 🥺

**Reply by poire-z** [reply]
> user style tweak, to put in the folder and activate it by tap in another similar book 

You can do that yourself: copy paste my suggestion in a file with suffix .css (for example: `kill_bad_indentation.css`, and put it in koreader/styletweaks/  (which should already exists),
After a restart of KOReader, in should appear under:
![image](https://github.com/koreader/koreader/assets/24273478/b3c6ba00-e140-436e-b72d-3974d58d18e4)

**Reply by mergen3107** [reply]
If it screws up too badly, use a negative text-indent for calibre4 :D

**Reply by ghost** [reply]
> > user style tweak, to put in the folder and activate it by tap in another similar book 
> 
> You can do that yourself: copy paste my suggestion in a file with suffix .css (for example: `kill_bad_indentation.css`, and put it in koreader/styletweaks/  (which should already exists),
> After a restart of KOReader, in should appear under:
> ![image](https://github.com/koreader/koreader/assets/24273478/b3c6ba00-e140-436e-b72d-3974d58d18e4)
> 

Today I found another problem with that (show on attached photos)

When there is a handwriting letter in italic text that problem with too big indentation shows up again. What do I need to fix in our own style tweak?


![IMG_20240521_181046.jpg](https://github.com/koreader/koreader/assets/168462969/4a2e3cb6-23bc-43e2-8ab5-6737e9f6bf93)


![IMG_20240521_181102.jpg](https://github.com/koreader/koreader/assets/168462969/c1e49f1b-e766-49dd-b2ad-19002115dca5)

Update: I just added the same sentence with calibre16 and it solved the problem. Thanks

**Reply by mergen3107** [reply]
Welcome to the css TweakTown :D
